### PR TITLE
misc small changes from ether

### PR DESCRIPTION
**Pull request by karenetheridge**
Some small fixes to docs and style that I made locally while stepping through code while debugging...  these should be pretty uncontroversial changes I think. :)

### CODE REVIEW

File: lib/JSON/Validator.pm [hunk 1]
```diff
@@ -300,22 +300,12 @@ sub _load_schema_from_text {
   return Mojo::JSON::decode_json($$text) if $$text =~ /^\s*\{/s;
 
   # YAML
-  $visit = sub {
```
Comment (jhthorsen) [inline lib/JSON/Validator.pm]: You can remote `my $visit` in this sub as well.

File: lib/JSON/Validator.pm [hunk 2]
```diff
@@ -875,10 +864,11 @@ sub _validate_type_object {
       : $self->_validate($data, $path, $schema->{then} // {});
   }
 
-  my $coerce = $self->{coerce}{defaults};
-  while (my ($k, $r) = each %{$schema->{properties}}) {
+  my %rules;
+  for my $k (keys %{$schema->{properties}}) {
```
Comment (jhthorsen) [inline lib/JSON/Validator.pm]: Why did you change from `while` to `for` here?

File: lib/JSON/Validator.pm [hunk 3]
```diff
@@ -887,11 +877,12 @@ sub _validate_type_object {
     }
   }
 
-  while (my ($p, $r) = each %{$schema->{patternProperties} || {}}) {
+  for my $p (keys %{$schema->{patternProperties} || {}}) {
```
Comment (jhthorsen) [inline lib/JSON/Validator.pm]: Why did you change from `while` to `for` here?

File: lib/JSON/Validator.pm [hunk 4]
```diff
@@ -904,7 +895,7 @@ sub _validate_type_object {
     return E $path, [object => additionalProperties => join '/', @k];
   }
 
-  for my $k (sort keys %required) {
+  for my $k (sort uniq @{$schema->{required} || []}) {
```
Comment (jhthorsen) [inline lib/JSON/Validator.pm]: 👍

File: lib/JSON/Validator.pm (hunk 2 above)
Comment (karenetheridge) [reply lib/JSON/Validator.pm]: I explained that in the comment for commit  46c2b23.

**Comment by jhthorsen** [comment]
I'll merge it even though I don't fully agree on the while/for changes. (Have to make a new release)
